project2/screens/search.js

```diff
@@ -1,131 +1,72 @@
 import React from "react";
-import { Button, FlatList, StyleSheet, Text, TouchableOpacity, TextInput, View } from "react-native";
-import Constants from 'expo-constants'
+import { FlatList, Text, TextInput, View } from "react-native";
 import { searchMovies, getMoviePage } from "../API";
-
-
-const Row = props => (
-    <TouchableOpacity onPress={() => props.selected(props.id)}>
-        <View style={styles.row}>
-            <Text style={styles.listText}>{props.title}</Text>
-        </View>
-    </TouchableOpacity>
-)
-const renderItem = selected => obj => <Row selected={selected}{...obj.item} />
-
-export default class search extends React.Component {
-    static navigationOptions = {
-        title: "Search",
-        headerTitleStyle: { color: "#AAAAAF", }
-    };
-
-    state = {
-        searchString: "",
-        movies: "",
-        totalResults: 0,
-        totalPages: 0,
-        nextPage: 0,
-    };
-
-    getMovies = async (searchString) => {
-        const results = await searchMovies(`${searchString}`)
-        console.log(results)
-
-        if (results !== undefined) {
-            this.setState({ movies: results["result"] })
-            this.setState({ totalResults: results["totalResults"] })
-            this.setState({ totalPages: results["totalPages"] })
-
-            if (results["totalPages"] > 1) {
-                this.setState({ nextPage: 2 })
-            }
-        }
-    }
-
-    componentDidUpdate(prevProps, prevState) {
-        if (this.state.searchString !== prevState.searchString) {
-            this.getMovies(this.state.searchString)
+import SearchListItem from "./Search/ListItem";
+import s from "../Styles";
+
+const Search = () => {
+  const [searchString, setSearchString] = React.useState("");
+  const [movies, setMovies] = React.useState([]);
+  const [totalResults, setTotalResults] = React.useState(0);
+  const [totalPages, setTotalPages] = React.useState(0);
+  const [nextPage, setNextPage] = React.useState(0);
+
+  React.useEffect(() => {
+    const getMovies = async () => {
```

Comment:
Again, as iwth the Movie compoent, we move the async function into the `useEffect` callback, it goes to search for movies, then sets the state based on the response.